File: project-euler/many-primes.clj

; The sum of the primes below 10 is 2 + 3 + 5 + 7 = 17.

; Find the sum of all the primes below two million.

(defn prime? [n]
  (if (even? n) false
      (let [root (num (int (Math/sqrt n)))]
	(loop [i 3]
	  (if (> i root) true
	      (if (zero? (mod n i)) false
		  (recur (+ i 2))))))))

(defn primes [n]
  (let [nums (loop [i 2 acc []]
	       (if (> i n) acc
		   (recur (inc i) (cons i acc))))]
    (concat (filter prime? nums) [2])))

;(println (reverse (primes 4000000)))
(primes 10000000)

;(println (primes 100))

File: project-euler/many-primes.js

var async = require('async');

var isPrime = function (n, callback) {
    if (n%2 == 0) {
	callback(false);
	return;
    }else{
	var root = Math.sqrt(n);
	for (var i=3; i<=root; i += 2) {
	    if (n%i == 0) {
		callback(false);
		return;
	    }
	}
    }
    callback(true);
}

var primes = function (n, callback) {
    var acc = new Array();

    for (var i=2; i<n; acc.push(i++));

    var fs = require('fs'), path = require('path');

    
    async.filter(acc, isPrime, function(results){
	results.unshift(2);
	callback(results);
	//console.log(results);
    });
}

primes(10000000, function (result) {
    //console.log(result);
});